### PR TITLE
Remove py27 support

### PR DESCRIPTION
**Pull request by Natim**

### CODE REVIEW

**Comment by Natim** [comment]
@leplatrem I have no Idea why the ETag contains twice ``""""`` probably I did something wrong but it doesn't looks obvious to me, any idea?

File: tests/plugins/test_history.py [hunk 1]
```diff
@@ -333,7 +333,7 @@ def test_full_deletion(self):
 
     def test_partial_deletion(self):
         resp = self.app.get('/buckets/bid/history', headers=self.headers)
-        before = resp.headers['ETag']
+        before = int(resp.headers['ETag'])
```
Comment (Natim) [inline tests/plugins/test_history.py]: @leplatrem any idea why it was working before?

Apparently:

```
./kinto/core/resource/schema.py:
  245 :     _before = QueryField(colander.Integer())
```

File: CHANGELOG.rst [hunk 2]
```diff
@@ -3,10 +3,13 @@ Changelog
 
 This document describes changes between each past release.
 
-5.4.0 (unreleased)
+6.0.0 (unreleased)
 ------------------
 
-- Nothing changed yet.
+**Internal changes**
```
Comment (leplatrem) [inline CHANGELOG.rst]: I would list those as **Breaking changes**

**Comment by leplatrem** [comment]
In many places we do something like that:
```py
headers = self.headers.copy()
headers.update({'Etag': '"123"'})
```

With Python3, now we can do:
```py
headers = {**self.headers, 'Etag': '"123"'}
```
or
```py
headers = {**self.headers, **get_user_headers('mat')}
```

**Comment by Natim** [comment]
I started that but it fails with a syntax error: ``self.aaron_headers = {**self.headers, **get_user_headers('aaron'}}``

**Comment by leplatrem** [comment]
We could also remove deprecated stuff like:
* `class Toto(object)`
* `super(Toto, self).__init__(*args, **kwargs)`
* String formatting with `%`

**Comment by Natim** [comment]
Can we do it in multiple PR?

**Comment by leplatrem** [comment]
> Can we do it in multiple PR?

Why not!

There is also the `raise from` for the backend errors

**Comment by Natim** [comment]
> There is also the raise from for the backend errors

Can you elaborate on that?

**Comment by leplatrem** [comment]
>>    There is also the raise from for the backend errors
>
> Can you elaborate on that?

I think we can do `raise BackendError(original=e) from e`  instead of just `raise BackendError(original=e)` to keep a proper stacktrace

**Comment by leplatrem** [comment]
Nice! 

Are there some flake8 options to prevent us from using deprecated features of python 3? (like `"hello %s" % name`?

**Comment by Natim** [comment]
I don't think %s is a deprecated feature of Python3 to be honest :)

**Comment by leplatrem** [comment]
> I don't think %s is a deprecated feature of Python3 to be honest :)

Saw that here:
*However, the % operator is still supported; it will be deprecated in Python 3.1 and removed from the language at some later time.)*
 https://docs.python.org/3/whatsnew/3.0.html#pep-3101-a-new-approach-to-string-formatting

File: tests/test_views_buckets.py [hunk 3]
```diff
@@ -193,7 +193,7 @@ def setUp(self):
                                MINIMALIST_RECORD,
                                headers=self.headers)
         record_id = r.json['data']['id']
-        self.record_url = self.collection_url + '/records/%s' % record_id
+        self.record_url = self.collection_url + '/records/{record_id}'.format(record_id=record_id)
```
Comment (leplatrem) [inline tests/test_views_buckets.py]: When there's only one you can do `"/records/{}".format(record_id)`

Comment (Natim) [inline tests/test_views_buckets.py]: Yes but is it a good idea to do thatM

Comment (leplatrem) [inline tests/test_views_buckets.py]: For that particular case I guess it is (may have to judge them one by one)

**Comment by Natim** [comment]
Things to do to move to the next `format` calls:

**Kinto**

[ ] kinto/__main__.py
[ ] kinto/authorization.py
[ ] kinto/config/__init__.py
[ ] kinto/core/__init__.py
[ ] kinto/core/authentication.py
[ ] kinto/core/authorization.py
[ ] kinto/core/cache/postgresql/__init__.py
[ ] kinto/core/initialization.py
[ ] kinto/core/logs.py
[ ] kinto/core/permission/memory.py
[ ] kinto/core/permission/postgresql/__init__.py
[ ] kinto/core/resource/__init__.py
[ ] kinto/core/resource/viewset.py
[ ] kinto/core/scripts.py
[ ] kinto/core/statsd.py
[ ] kinto/core/storage/exceptions.py
[ ] kinto/core/storage/memory.py
[ ] kinto/core/storage/postgresql/__init__.py
[ ] kinto/core/storage/postgresql/client.py
[ ] kinto/core/storage/testing.py
[ ] kinto/core/testing.py
[ ] kinto/core/utils.py
[ ] kinto/core/views/batch.py
[ ] kinto/core/views/errors.py
[ ] kinto/core/views/heartbeat.py
[ ] kinto/plugins/admin/views.py
[ ] kinto/plugins/default_bucket/__init__.py
[ ] kinto/plugins/history/listener.py
[ ] kinto/views/permissions.py
[ ] kinto/views/records.py

**Tests**

[ ] tests/core/resource/test_events.py
[ ] tests/core/resource/test_filter.py
[ ] tests/core/resource/test_object_permissions.py
[ ] tests/core/resource/test_record.py
[ ] tests/core/support.py
[ ] tests/core/test_authentication.py
[ ] tests/core/test_storage.py
[ ] tests/core/test_utils.py
[ ] tests/core/test_views_batch.py
[ ] tests/core/test_views_hello.py
[ ] tests/plugins/test_default_bucket.py
[ ] tests/plugins/test_history.py
[ ] tests/plugins/test_quotas.py
[ ] tests/support.py
[ ] tests/test_main.py
[ ] tests/test_views_collections.py
[ ] tests/test_views_collections_schema.py
[ ] tests/test_views_records.py

File: CHANGELOG.rst [hunk 4]
```diff
@@ -19,6 +19,7 @@ This document describes changes between each past release.
 
 - Update the upsert query to use an INSERT or UPDATE on CONFLICT behavior (fixes #1055)
 - Remove pypy supports. (#1049)
+- Remove Python 2.7 support. (#1050)
```
Comment (leplatrem) [inline CHANGELOG.rst]: This is a big breaking change!

**Comment by glasserc** [comment]
Per @Natim on IRC, those of you following along at home should do `make maintainer-clean` in your kinto source tree to rebuild a venv with Python 3.

**Pull request merged**